### PR TITLE
Network Sensor

### PR DESCRIPTION
**Pull request by adityaguptakd**
I am not sure I fully understand how the network sensor should be implemented, is each actual sensor associated with a network sensor and then network sensor acts to store values and communicate? Anyways this could be all wrong.

### CODE REVIEW

File: ceaos/sensors/nwsensor.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
+from .sensor_definition import Sensor
+from ..loggers.InfluxDB import InfluxDBLogger
+
+
+class Network_Sensor(Sensor):
```
Comment (nandantumu) [inline ceaos/sensors/nwsensor.py:5]: Can you rename this `NetworkSensor`?

File: ceaos/sensors/nwsensor.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    def __init__(self, sensor=None, ):
+        self.curr_value = None
+        self.logger = InfluxDBLogger()
```
Comment (nandantumu) [inline ceaos/sensors/nwsensor.py:8]: You should allow the logger to be passed into the constructor

File: ceaos/sensors/nwsensor.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        # I think each Network_Sensor is essentially
+        # attached to an actual sensor
+        self.sensor = sensor
```
Comment (nandantumu) [inline ceaos/sensors/nwsensor.py:11]: You don't need this! Remember that the interface for a Sensor when initialized will create a NotImplementedError

File: ceaos/sensors/nwsensor.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+    def read_value(self):
+        return self.sensor.read_value()
```
Comment (nandantumu) [inline ceaos/sensors/nwsensor.py:14]: Just store the value inside the object

**Review by nandantumu — CHANGES_REQUESTED** [review]
Almost there, Just remember to add some tests.

File: ceaos/sensors/nwsensor.py (hunk 4 above)
Comment (adityaguptakd) [reply ceaos/sensors/nwsensor.py:14]: what do you mean by that? isn't each network sensor associated with a "real sensor" so the network sensor read_value is essentially just calling another sensors read value? Or by store do you mean create a attribute that also stores the value and returns it?

File: ceaos/sensors/nwsensor.py (hunk 3 above)
Comment (adityaguptakd) [reply ceaos/sensors/nwsensor.py:11]: Not sure I follow, for each network sensor, don't we need to pass in an actual sensor hence, self.sensor = sensor?

Comment (nandantumu) [reply ceaos/sensors/nwsensor.py:11]: Nope, There's no need for this, we can just treat them all the same. We may want to keep track of the datatype, but that's it.

File: ceaos/sensors/nwsensor.py (hunk 4 above)
Comment (nandantumu) [reply ceaos/sensors/nwsensor.py:14]: No need for a real sensor, just a value attribute is fine

**Pull request closed without merging**